src/read_pool.rs

```diff
@@ -140,14 +152,29 @@ impl ReadPoolHandle {
                     CommandPri::Normal => None,
                     CommandPri::Low => Some(2),
                 };
-                let extras = Extras::new_multilevel(task_id, fixed_level);
-                let task_cell = TaskCell::new(
-                    TrackedFuture::new(async move {
-                        f.await;
-                        running_tasks.dec();
-                    }),
-                    extras,
-                );
+                let mut extras = Extras::new_multilevel(task_id, fixed_level);
+                extras.set_metadata(group_meta.clone());
+                let task_cell = if let Some(resource_ctl) = resource_ctl {
+                    TaskCell::new(
+                        TrackedFuture::new(ControlledFuture::new(
```

Comment:
What do you think about integrating resource controller to the `Runner`? Then, we needn't another `group_meta` clone for each future.

Reply:
I don't find a good way to do so. Because typically, when scheduling a task, the system may need to generate a priority value for it and update some internal state based on the task's consuming resources. It's hard to do the second step in the runner. In this implement, as we only track the cpu time usage, it's possible to support in the runner. In the future, if we want to track more resources, e.g. reading bytes, it's seems hard to do it in the runner.